File: c++/euler_007.cpp
/*
 * Problem 7: 10001st isPrime
 * 
 * By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
 * What is the 10 001st prime number?
 */

#include <iostream>
#include <math.h>

using namespace std;

bool isPrime(int num)
{
	if(num % 2 == 0)
		return false;
	for(int i = 3; i * i <= num; i += 2)
		if(num % i == 0)
			return false;
	return true;
}

int main(void)
{
	const int max = 10001;
	short count = 1;
	int i = 1;
	while(count < max)
	{
		if(isPrime(++i))
			count++;
	}
	cout << i << endl;
}

File: python/euler_007.py
"""
Problem 7: 10001st prime
"""

def is_prime(num):
	if num > 2:
		if num % 2 == 0:
			return False
		for i in range(3, int(num ** 0.5) + 1, 2):
			if num % i == 0:
				return False
		return True	
	else:
		if num <= 1:
			return False
		if num == 2:
			return True

if __name__ == "__main__":
	count = 0
	n = 1
	while count < 10001:
		n += 1
		if is_prime(n):
			count += 1
	print n